src/util/helperUtil.ts

```diff
@@ -80,3 +81,60 @@ export function clearDataSavedInGlobalState(context: vscode.ExtensionContext) {
 	context.globalState.update('selectedProject', undefined);
 }
 
+/**
+ * Traverse upwards to check for target/build directory and its corresponding sibling build file (pom.xml or build.gradle).
+ * @param filePath The file path to start the search from.
+ * @returns true if sibling file found, false otherwise.
+ */
+export async function checkSiblingFilesInTargetOrBuildParent(filePath: string): Promise<boolean> {
+
+	try {
+		let currentDir = path.dirname(filePath);  // Start with the directory of the provided file
+
+		// Traverse upwards through the directories
+		while (currentDir !== path.parse(currentDir).root) {
+			let outputDir: string | undefined;
+			let siblingFilePath: string | undefined;
+
+			// Determine the file type to look for (either pom.xml or build.gradle)
+			if (filePath.endsWith('pom.xml')) {
+				outputDir = 'target';  // For Maven projects, look for target directory
+				siblingFilePath = path.join(currentDir, 'pom.xml');
+			} else if (filePath.endsWith('build.gradle') || filePath.endsWith('settings.gradle')) {
+				outputDir = 'build';  // For Gradle projects, look for build directory
+				siblingFilePath = path.join(currentDir, 'build.gradle');
+			} else {
+				console.log("Invalid file type. Only 'pom.xml' or 'build.gradle' are supported.");
+				return false;
+			}
+
+			// Ensure that targetDir and siblingFilePath are assigned before proceeding
+			if (!outputDir || !siblingFilePath) {
+				console.error("Error: targetDir or siblingFilePath not properly assigned.");
+				return false;
+			}
+
+			// Check if the target/build directory exists
+			const currentTargetDir = path.join(currentDir, outputDir);
+			if (fs.existsSync(currentTargetDir) && fs.statSync(currentTargetDir).isDirectory()) {
+
+				// Check if the expected sibling file (pom.xml or build.gradle) exists
+				if (fs.existsSync(siblingFilePath) && fs.statSync(siblingFilePath).isFile()) {
+					console.log(`Found sibling ${filePath.endsWith('pom.xml') ? 'pom.xml' : 'build.gradle'} in directory: ${currentTargetDir}`);
```

Comment:
With my recommended code changes above, you could use the `siblingFileType` var on this line:
```suggestion
					console.log(`Found sibling ${siblingFileType} in directory: ${currentTargetDir}`);
```

Reply:
addressed